pkg/storage/storage.go

```diff
@@ -66,7 +64,7 @@ type PaginationOptions struct {
 // it uses DefaultPageSize.
 func NewPaginationOptions(ps int32, contToken string) PaginationOptions {
 	pageSize := DefaultPageSize
-	if ps != 0 {
+	if ps <= 0 {
```

Comment:
```suggestion
	if ps > 0 {
```

pagination options can not be negative.